engines/stark/ui/world/gamescreen.cpp

```diff
@@ -133,4 +138,15 @@ void GameScreen::notifyDiaryEntryEnabled() {
 	_topMenu->notifyDiaryEntryEnabled();
 }
 
+void GameScreen::pauseGame(bool pause) {
+	if (StarkGlobal->getLevel()) {
+		StarkGlobal->getLevel()->onEnginePause(pause);
+	}
+	if (StarkGlobal->getCurrent()) {
+		StarkGlobal->getCurrent()->getLevel()->onEnginePause(pause);
+		StarkGlobal->getCurrent()->getLocation()->onEnginePause(pause);
+	}
+	g_engine->_mixer->pauseAll(pause);
```

Comment:
I am still not sure about how the pausing should work. Based on my observation, the only thing that actually does something in `onEnginePause()` is the animation. Yet the menu cannot be opened in most important animations anyway. For other simple animation such as when clicking April, it is naturally paused by the menu in the previous code. My guess is that the animation will naturally be paused when it is not rendered. So I really don't see the importance of pausing other things when the menu is opened. Is there a location where I can actually see that not pausing the game in the menu will cause problems?

As for pausing sounds, calling the `pauseAll()` of the mixer seems not to affect newly added sounds later. I see that sounds will automatically call `stop()` when a location is unloaded, so everything should be right..ish?

Reply:
> I am still not sure about how the pausing should work. Based on my observation, the only thing that actually does something in onEnginePause() is the animation. Yet the menu cannot be opened in most important animations anyway. For other simple animation such as when clicking April, it is naturally paused by the menu in the previous code. My guess is that the animation will naturally be paused when it is not rendered. So I really don't see the importance of pausing other things when the menu is opened. Is there a location where I can actually see that not pausing the game in the menu will cause problems?

You can go to location "50 00" to see things go wrong when coming back from the menu without calling `onEnginePause`.

> As for pausing sounds, calling the pauseAll() of the mixer seems not to affect newly added sounds later. I see that sounds will automatically call stop() when a location is unloaded, so everything should be right..ish?

Bypassing the entity system for sounds does not seem to follow the design of the engine to me. The entities can implement handlers to react to the events that may happen, let's use them.